### DISCUSSION TITLE
Select from list of choices OR custom entry (new to GUIs)

### DISCUSSION THREAD

**Discussion by cbc02009** [emilk/egui#1610] [Q&A]
Hello,

I'm working on a utility to help me sort/organize my media collection. I'm brand new to any sort of GUI work, so this is very much a learning exercise. Basically you enter a directory, and the program globs all the files and tries to parse important info from the file name (title, author, etc). I have a set of regex to try and parse filenames into book/movie titles, and I'd like to present the user with a list of the regex results, plus a box for custom entry like this:
![image](https://user-images.githubusercontent.com/39525900/167446972-8374d2c2-de4b-4da2-8664-f86f3fd35b07.png)

Any of the regex results may not be present if the regex didn't find anything. (if that over complicates things I can just leave empty radio buttons instead, or grayed out ones if I can figure that out.) The data is stored in a vec of structs, with back and next moving up and down the vec to display info from the struct.

I basically have two questions: 

First I'm not really sure how to use radio buttons in egui. Are there any good examples for them in code somewhere?

Once I figure out how to use radio buttons, how do I handle the custom entry field? I was thinking of having a radio button being selected populate the custom entry text box with that value, and the program just reads whatever is in the text box as the final value when the user clicks next (not the value from the selected radio button). Is this the simplest way to handle this, or is there a better way?

Any help would be greatly appreciated!

**Comment by willser** [comment]
Hi,you could take a look at [demo](https://www.egui.rs/).There a demo named `Misc Demos` maybe helpful for you.

**Reply by cbc02009** [reply]
Thank you for the suggesetion. With that I was able to get started, and the radio values now show up, but I can't click on them at all. do radio values only work with enums?

**Comment by emilk** [comment]
Have you checked the documentation? https://docs.rs/egui/latest/egui/widgets/struct.RadioButton.html

**Reply by cbc02009** [reply]
Sorry for the delayed reply, I got super busy at work. I have looked at the documentation, but I'm not sure I understand it. I *think* I've implemented it correctly, but clicking on the buttons doesn't seem to do anything.

```rust
ui.label("Titles:");
let mut title_selection = String::new();
for title in &b.binfo.titles {
    ui.radio_value(&mut title_selection, title.to_owned(), &*title);
}

ui.label(&*title_selection);
```

This is the code I'm working with right now. b.binfo.titles is a `HashSet<String>` of potential title matches from a passed file name. I'm making a radio_value for each one to let the user select which is correct. The last label is just for debugging so I can see if anything happens when the radio button is clicked, but the last label is never appearing.

Like I said, I'm fairly new to rust and 100% new to GUIs, so it's possible that it's just some stupid syntax mistake I'm making.

**Reply by emilk** [reply]
You need to store the selected value somewhere (`title_selection`). Now you are saying to egui "There is no selection, but if someone clicks on something, then show that selection for exactly one frame, and then forget about it". See e.g. https://github.com/emilk/egui/blob/master/examples/hello_world/src/main.rs

**Reply by cbc02009** [reply]
oh, duh, because it's immediate mode. Sorry for the newbie mistake. Thanks!
